### PR TITLE
Bump versions to .NET 6 and upgrade C# code to v10

### LINKED ISSUE #27127: Example for top level statements
Is there any plan for documentation that shows how this is best achieved using top level statements?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a4eb702a-d75f-f780-aef1-2e858e14258f
* Version Independent ID: 9a183e07-d5fd-1de9-796d-3bc77f07e033
* Content: [Use dependency injection in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage)
* Content Source: [docs/core/extensions/dependency-injection-usage.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/dependency-injection-usage.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Bump versions to .NET 6 and upgrade C# code to v10

> I really wish that the triple colon syntax for our MD code blocks would allow for comma-delimited `id` values!
>
> ```md
> :::code source="../path/to/file.cs" id="one,two":::
> ```
>
> This would avoid the `range` syntax, which is not future-proof. ☹️

Part of #26777, relying on GHA output from #26800, also fixes #27127

## Previews

Lots of content relied on `range`, ported what I could to `id` and fixed the remaining bits:

- [docs/core/extensions/configuration-providers.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers?branch=pr-en-us-27051)
- [docs/core/extensions/configuration.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/configuration?branch=pr-en-us-27051)
- [docs/core/extensions/console-log-formatter.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/console-log-formatter?branch=pr-en-us-27051)
- [docs/core/extensions/custom-configuration-provider.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/custom-configuration-provider?branch=pr-en-us-27051)
- [docs/core/extensions/custom-logging-provider.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/custom-logging-provider?branch=pr-en-us-27051)
- [docs/core/extensions/dependency-injection-guidelines.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines?branch=pr-en-us-27051)
- [docs/core/extensions/dependency-injection-usage.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-usage?branch=pr-en-us-27051)
- [docs/core/extensions/dependency-injection.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection?branch=pr-en-us-27051)
- [docs/core/extensions/high-performance-logging.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging?branch=pr-en-us-27051)
- [docs/core/extensions/localization.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/localization?branch=pr-en-us-27051)
- [docs/core/extensions/logger-message-generator.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator?branch=pr-en-us-27051)
- [docs/core/extensions/logging-providers.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/logging-providers?branch=pr-en-us-27051)
- [docs/core/extensions/logging.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/logging?branch=pr-en-us-27051)
- [docs/core/extensions/options-library-authors.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/options-library-authors?branch=pr-en-us-27051)
- [docs/core/extensions/options.md](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/options?branch=pr-en-us-27051)

### CODE REVIEW

**Comment by BillWagner** [comment]
I looked briefly at the file list. Should we update the `.editorconfig` with these settings as well?

File: docs/core/extensions/logging-providers.md [hunk 1]
```diff
@@ -188,10 +188,10 @@ This provider only logs when the project runs in the Azure environment.
 
 #### Azure log streaming
 
-Azure log streaming supports viewing log activity in real time from:
+Azure log streaming supports viewing log activity in real-time from:
 
 - The app server
-- The web server
+- The webserver
```
Comment (tdykstra) [inline docs/core/extensions/logging-providers.md:194]: Two words seems correct to me.

**Review by tdykstra — APPROVED** [review]
LGTM

File: docs/core/extensions/logging-providers.md (hunk 1 above)
Comment (IEvangelist) [reply docs/core/extensions/logging-providers.md:194]:
```suggestion
- The web server
```

**Pull request merged**